File: 0696-count-binary-substrings/0696-count-binary-substrings.py
class Solution:
    def countBinarySubstrings(self, s: str) -> int:
        res = 0
        first = 1
        i = 1
        while i < len(s) and s[i] == s[i - 1]:
            first += 1
            i += 1
        while i < len(s):
            second = 1
            i += 1
            while i < len(s) and s[i] == s[i - 1]:
                second += 1
                i += 1
            res += min(first, second)
            first = second
        return res

File: 0696-count-binary-substrings/0696-count-binary-substrings.cpp
class Solution {
public:
    int countBinarySubstrings(string s) {
        int res = 0;
        int first = 0;
        int second = 1;
        for(int i = 1; i < s.length(); i++)
        {
            if(s[i] == s[i - 1])
            {
                second++;
            }
            else
            {
                res += min(first, second);
                first = second;
                second = 1;
            }
        }
        return res + min(first, second);
    }
};